src/res_core.ml

```diff
@@ -1020,10 +1020,6 @@ let parseTemplateConstant ~prefix (p : Parser.t) =
   match p.token with
   | TemplateTail txt ->
     Parser.next p;
-    let txt =
-      if p.mode = ParseForTypeChecker then parseTemplateStringLiteral txt
```

Comment:
One difference with normal strings is that normal strings do:
```ml
      if p.mode = ParseForTypeChecker then Pconst_string (s, Some "js")
      else Pconst_string (s, None)
```
but I guess here we don't want to change the prefix.

CC @IwanKaramazow

Reply:
This is left in case of `#foo`:

```ml
            let text =
              if p.mode = ParseForTypeChecker then parseStringLiteral text
              else text
            in
```
and several other cases. Is that intentional

Also, there's the case of `TemplatePart` that needs looking into, and is not changed yet.

Reply:
Right, I guess for poly-variants with string literals: `#"string here"`. The question is if what happens when we have unicode and/or escape codes in here.

Reply:
> Right, I guess for poly-variants with string literals: `#"string here"`. The question is if what happens when we have unicode and/or escape codes in here.

Does the compiler care if there's unicode in the name of a poly variant? I guess it does not look at the string at all.
Is the question about when printing a type error?

Reply:
This might be outside of the scope of this PR, but the question is what we want here:
<img width="1325" alt="image" src="https://user-images.githubusercontent.com/10634678/174455549-650910e1-dce1-44ff-b014-e67a63a3831d.png">